src/Agent.cs

```diff
@@ -37,18 +37,25 @@ public int MessageCount
 
     public Agent(String apiKey)
     {
+      if(string.IsNullOrEmpty(apiKey))
+        throw new ArgumentException("api key was null or missing", apiKey);
+
       Enabled = true;
-      Synchronous = false;
       _collector = new Collector(apiKey);
+
+      _validateMetric = new Regex(
+                                  @"^([\d\w\-_]+\.)*[\d\w\-_]+$",
+                                  RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.ExplicitCapture
```

Comment:
They, together, are one argument, so I think having them grouped is better.  It may make sense to pull them out into a var... but really, this whole thing should be in a static initializer, assuming regex is thread safe.